vexim/siteaddsubmit.php

```diff
@@ -79,12 +73,13 @@
     $pophomepath = $domainpath . "/" . $_POST['localpart'];
   }
 //Gah. Transactions!! -- GCBirzan
-  if ((validate_password($_POST['clear'], $_POST['vclear'])) &&
-    ($_POST['type'] != "alias")) {
-      if (!password_strengthcheck($_POST['clear'])) {  
-        header ("Location: site.php?weakpass={$_POST['domain']}");
-        die;
-      }
+  if ($_POST['type'] !== "alias") {
+    if ($_POST['type'] === "local" && validate_password($_POST['clear'], $_POST['vclear'])) {
```

Comment:
If `validate_password` fails, this will create the domain, without even checking if password is strong.

Line 140 should be changed to simple `else`, and lines 153-154 should be moved below this condition.